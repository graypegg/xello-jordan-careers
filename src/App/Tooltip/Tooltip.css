.Tooltip__container {
  display: inline-block;
  position: relative;
  cursor: pointer;
}

.Tooltip__tooltip {
  position: absolute;
  top: 100%;
  left: 50%;
  padding: 10px 14px 10px 18px;
  border-radius: 5px;
  background: #FFF;
  color: #656565;
  font-size: 0.75rem;
  box-shadow: 0px 0px 10px rgba(30, 30, 40, 0.3);
  min-width: 170px;
  margin-top: 8px;
  pointer-events: none;
  letter-spacing: 0.02rem;
  line-height: 16px;
}

.Tooltip__tooltip strong {
  display: block;
  font-size: 0.85rem;
  margin-bottom: 3px;
}

.Tooltip__tooltip::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  border-style: solid;
  position: absolute;
  top: 0;
  left: 50%;
  transform: translate(-50%, -100%);
  border-width: 0 7px 8px 7px;
  border-color: transparent transparent #FFF transparent;
}

.Tooltip__tooltip::after {
  content: '';
  background: var(--primary-light);
  display: block;
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 8px;
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.Tooltip__container .Tooltip__tooltip {
  opacity: 0;
  transform-origin: top center;
  transform: perspective(500px) translate(-50%, 0) rotate3d(0, 1, 0, 90deg);
  transition: all 0.3s cubic-bezier(0.19, 1, 0.22, 1);
}

.Tooltip__container:hover .Tooltip__tooltip {
  opacity: 1;
  transform: perspective(500px) translate(-50%, 0) rotate3d(0, 0, 0, 45deg);
}
