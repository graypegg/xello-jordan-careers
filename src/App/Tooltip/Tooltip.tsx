import * as React from 'react'

import './Tooltip.css'

interface ITooltipProps {
  children?: any
  content: string,
  title?: string,
  active?: boolean
}

export default function Tooltip (props: ITooltipProps): JSX.Element {
  return (
    <div className="Tooltip__container">
      { props.children }
      { props.active === undefined || props.active === true
        ? <div className="Tooltip__tooltip">
            {
              props.title
              ? <strong>{ props.title }</strong>
              : null
            }
            { props.content }
          </div>
        : null
      }
    </div>
  )
}