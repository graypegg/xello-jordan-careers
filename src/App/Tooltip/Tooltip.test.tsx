import * as React from 'react'
import { shallow } from 'enzyme'
import Tooltip from './Tooltip'

it('renders without crashing', () => {
  const wrapper = shallow(<Tooltip title="xyz" content="abc">ailurus fulgens</Tooltip>)
  expect(wrapper).toMatchSnapshot()
})

it('hides when active is false', () => {
  const wrapper = shallow(<Tooltip title="xyz" content="abc" active={false}>ailurus fulgens</Tooltip>)
  expect(wrapper.find('.Tooltip__tooltip').length).toBe(0);
})

it('shows when active is true', () => {
  const wrapper = shallow(<Tooltip title="xyz" content="abc" active={true}>ailurus fulgens</Tooltip>)
  expect(wrapper.find('.Tooltip__tooltip').length).toBe(1);
})

it('can show with out a title', () => {
  const wrapper = shallow(<Tooltip content="abc">ailurus fulgens</Tooltip>)
  expect(wrapper).toMatchSnapshot()
})