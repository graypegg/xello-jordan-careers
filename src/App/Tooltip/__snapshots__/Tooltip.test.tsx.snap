// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can show with out a title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tooltip
    content="abc"
  >
    ailurus fulgens
  </Tooltip>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "ailurus fulgens",
        <div
          className="Tooltip__tooltip"
        >
          abc
        </div>,
      ],
      "className": "Tooltip__container",
    },
    "ref": null,
    "rendered": Array [
      "ailurus fulgens",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            "abc",
          ],
          "className": "Tooltip__tooltip",
        },
        "ref": null,
        "rendered": Array [
          null,
          "abc",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "ailurus fulgens",
          <div
            className="Tooltip__tooltip"
          >
            abc
          </div>,
        ],
        "className": "Tooltip__container",
      },
      "ref": null,
      "rendered": Array [
        "ailurus fulgens",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              "abc",
            ],
            "className": "Tooltip__tooltip",
          },
          "ref": null,
          "rendered": Array [
            null,
            "abc",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tooltip
    content="abc"
    title="xyz"
  >
    ailurus fulgens
  </Tooltip>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "ailurus fulgens",
        <div
          className="Tooltip__tooltip"
        >
          <strong>
            xyz
          </strong>
          abc
        </div>,
      ],
      "className": "Tooltip__container",
    },
    "ref": null,
    "rendered": Array [
      "ailurus fulgens",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <strong>
              xyz
            </strong>,
            "abc",
          ],
          "className": "Tooltip__tooltip",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "xyz",
            },
            "ref": null,
            "rendered": "xyz",
            "type": "strong",
          },
          "abc",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "ailurus fulgens",
          <div
            className="Tooltip__tooltip"
          >
            <strong>
              xyz
            </strong>
            abc
          </div>,
        ],
        "className": "Tooltip__container",
      },
      "ref": null,
      "rendered": Array [
        "ailurus fulgens",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <strong>
                xyz
              </strong>,
              "abc",
            ],
            "className": "Tooltip__tooltip",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "xyz",
              },
              "ref": null,
              "rendered": "xyz",
              "type": "strong",
            },
            "abc",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
